src/electron/blocker.ts

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright (c) 2017-2019 Cliqz GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+// Type definition
+import { ipcMain } from 'electron';
+import { parse } from 'tldts';
+
+import { IBackgroundCallback, IMessageFromBackground } from '../../cosmetics';
+import Engine from '../engine/engine';
+import Request from '../request';
+
+/**
+ * Wrap `FiltersEngine` into a Electron-friendly helper class.
+ */
+export default class ElectronBlocker extends Engine {
+  public enableBlockingInSession(ses: Electron.Session) {
+    ses.webRequest.onBeforeRequest({ urls: ['<all_urls>'] }, this.onRequest);
+    ses.setPreloads(['./content.js']);
```

Comment:
Maybe this `setPreloads` could be called instead by the user of the adblocker? By calling `enableBlockingInSession(...)` you would enable network requests blocking + listening to content messages. If you also inject `content.js` (e.g.: from `examples/electron/index.ts`) you would get cosmetics. I think it might be easier this way.

Reply:
I don't think it would be a good idea, since users would need to know the exact path of the `content.js` and it's problematic.

Reply:
Sure, but the current way seems to be incompatible with bundling? Not sure how that's going to work since in the end the package exposes few single-files bundles. Is there any way we could inject the script as a string containing the code?

Reply:
We could probably create a temp file with the code and pass its path, but in my opinion it would be better to create a separate package for electron `@cliqz/adblocker-electron` and we could build the module with `tsc`.

Reply:
Agreed, separate packages seem like a better way to go in the future. I can setup this new architecture for the repo so that we can build on top then?

Reply:
Okay